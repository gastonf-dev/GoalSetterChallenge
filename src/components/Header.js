/**
 * @flow stric-local
 * @format
 */

import React from 'react';
import type {Node} from 'react';

import {ImageBackground, Platform, StyleSheet, Text, View} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

const IMAGE_HEADER = require('../assets/images/background.png');

type Props = {
  title: string,
  subtitle?: string,
};
const Header = ({title, subtitle}: Props): Node => {
  return (
    <>
      <ImageBackground source={IMAGE_HEADER} style={styles.imageHeader}>
        <Text style={styles.title}>{title}</Text>
        {subtitle && <Text style={styles.subtitle}>{subtitle}</Text>}
      </ImageBackground>
    </>
  );
};

export default Header;

var styles = StyleSheet.create({
  imageHeader: {
    resizeMode: 'cover',
    justifyContent: 'center',
    padding: 24,
  },
  title: {
    fontFamily: Platform.OS === 'ios' ? 'Avenir LT Std' : 'AvenirLTStd',
    fontWeight: 'bold',
    fontSize: 20,
    lineHeight: 23,
    textAlign: 'center',
    color: 'white',
  },
  subtitle: {
    fontFamily: Platform.OS === 'ios' ? 'Avenir LT Std' : 'AvenirLTStd',
    fontSize: 14,
    lineHeight: 16,
    textAlign: 'center',
    color: 'white',
    marginTop: 12,
  },
});
