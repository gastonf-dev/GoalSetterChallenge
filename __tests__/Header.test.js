import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../src/components/Header';
describe('<Header />', () => {
  test('render header with title and subtitle', () => {
    const PROPS = {
      title: 'Test title',
      subtitle:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin luctus ipsum nec sapien pretium iaculis. Vestibulum condimentum sapien vitae felis consequat maximus.',
    };
    const wrapper = renderer.create(<Header {...PROPS} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('render header with title only', () => {
    const PROPS = {
      title: 'Test title',
    };
    const wrapper = renderer.create(<Header {...PROPS} />);
    expect(wrapper).toMatchSnapshot();
  });
});
