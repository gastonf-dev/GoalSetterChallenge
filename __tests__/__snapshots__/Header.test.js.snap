// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render header with title and subtitle 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "justifyContent": "center",
      "padding": 24,
      "resizeMode": "cover",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Avenir LT Std",
        "fontSize": 20,
        "fontWeight": "bold",
        "lineHeight": 23,
        "textAlign": "center",
      }
    }
  >
    Test title
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Avenir LT Std",
        "fontSize": 14,
        "lineHeight": 16,
        "marginTop": 12,
        "textAlign": "center",
      }
    }
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin luctus ipsum nec sapien pretium iaculis. Vestibulum condimentum sapien vitae felis consequat maximus.
  </Text>
</View>
`;

exports[`<Header /> render header with title only 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "justifyContent": "center",
      "padding": 24,
      "resizeMode": "cover",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Avenir LT Std",
        "fontSize": 20,
        "fontWeight": "bold",
        "lineHeight": 23,
        "textAlign": "center",
      }
    }
  >
    Test title
  </Text>
</View>
`;
